src/ObjCRuntime/PlatformAvailability2.cs

```diff
@@ -92,6 +92,76 @@ internal AvailabilityBaseAttribute ()
 		public override string ToString ()
 		{
 			var builder = new StringBuilder ();
+#if NET
+			switch (AvailabilityKind) {
+			case AvailabilityKind.Introduced:
+				builder.Append ("[SupportedOSPlatform (\"");
+				break;
+			case AvailabilityKind.Obsoleted:
+				switch (Platform) {
+				case PlatformName.iOS:
+					builder.AppendLine ("#if __IOS__");
+					break;
+				case PlatformName.TvOS:
+					builder.AppendLine ("#if __TVOS__");
+					break;
+				case PlatformName.WatchOS:
+					builder.AppendLine ("#if __WATCHOS__");
+					break;
+				case PlatformName.MacOSX:
+					builder.AppendLine ("#if __MACOS__");
+					break;
+				case PlatformName.MacCatalyst:
+					builder.AppendLine ("#if __MACCATALYST__");
+					break;
+				}
```

Comment:
Maybe show an error/throw an exception if an unknown platform?

Reply:
not really needed if the 2nd one is present
IOW this one is conditional but the other is always executed and the platform can't change between them

Reply:
It's not that it's needed for input validation, it's that it's helpful when the next platform comes around. Much easier to figure out what to do with a "Unknown platform" error/exception than a random build error because the generated code was incorrect.